### DISCUSSION TITLE
Cannot connect to AWS Athena Using PyAthena

### DISCUSSION THREAD

**Discussion by nghinguyen157** [apache/superset#27543] [Q&A / Help] [answered]
Hi every one, 

I'm facing an issue when try to connect with Athena using PyAthena with the connection string (SQLALCHEMY URI): 

awsathena+rest://{aws_access_key_id}:{aws_secret_access_key}@athena.{region_name}.amazonaws.com/{schema_name}?s3_staging_dir={s3_staging_dir}

It's prompt this error in docker logs: 

superset.commands.database.exceptions.DatabaseTestConnectionUnexpectedError: [SupersetError(message="module 'sqlalchemy.types' has no attribute 'DOUBLE'", error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'Amazon Athena', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]

{s3_staging_dir} is s3 Bucket that has created to store AWS Athena query result.
{schema_name} is database name that being created in AWS Glue Crawler. 
{aws_access_key_id}:{aws_secret_access_key} : is AWS key pair belong to an IAM user which has appropriate permission.

Kindly suggest

**Comment by 14shreyansh** [comment]
@nghinguyen157 Did you find any solution for this? I am facing the same error.

**Comment by jgournet** [comment]
We got the same error.
Rolling back to our previous superset image, and it's working fine (built on the 9th of March) from superset docker image latest
=> might be an issue introduced with 3.1.1 image ?

**Comment by dustin-liu-bgl** [comment]
Hi team,
This is a serious issue, would you please help to look into this as soon as possible? thanks.

**Comment by jgournet** [comment]
Someone reported the same issue on slack too.
=> I created the ticket: https://github.com/apache/superset/issues/27559 instead of this discussion, in the hope of getting more traction.

**Comment by 14shreyansh** [comment]
It worked with PyAthena 3.0.0 version. The latest version of PyAthena seems to have problem.

**Comment by jgournet** [comment, marked as answer]
Looks like pyathena released 3.5.1 with a fix for this issue:
https://github.com/laughingman7743/PyAthena/releases/tag/v3.5.1
